miio/toiletlid.py

```diff
@@ -0,0 +1,147 @@
+        self.data = data
+
+    @property
+    def work_state(self) -> int:
+        """Device state code"""
+        return self.data["work_state"]
+
+    @property
+    def is_on(self) -> bool:
+        return self.work
+
+    @property
+    def work(self) -> bool:
+        """True if device is use on."""
+        return self.work_state != 1
+
+    @property
+    def filter_use_percentage(self) -> str:
+        """Filter percentage of remaining life"""
+        return "{}%".format(self.data["filter_use_flux"])
+
+    @property
+    def filter_remaining_time(self) -> int:
+        """Filter remaining life days"""
+        return self.data["filter_use_time"]
+
+    @property
+    def ambient_light(self) -> str:
+        """Ambient light color."""
+        return self.data["ambient_light"]
+
+    def __repr__(self) -> str:
+        return (
+            "<ToiletlidStatus work=%s, "
+            "state=%s, "
+            "ambient_light=%s, "
+            "filter_use_percentage=%s, "
+            "filter_remaining_time=%s>"
+            % (
+                self.work,
+                self.work_state,
+                self.ambient_light,
+                self.filter_use_percentage,
+                self.filter_remaining_time,
+            )
+        )
+
+
+class Toiletlid(Device):
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_TOILETLID_V1,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in AVAILABLE_PROPERTIES:
+            self.model = model
+        else:
+            self.model = MODEL_TOILETLID_V1
+
+    @command(
+        default_output=format_output(
+            "",
+            "Work: {result.work}\n"
+            "State: {result.work_state}\n"
+            "Ambient Light: {result.ambient_light}\n"
+            "Filter remaining: {result.filter_use_percentage}\n"
+            "Filter remaining time: {result.filter_remaining_time}\n",
+        )
+    )
+    def status(self) -> ToiletlidStatus:
+        """Retrieve properties."""
+        properties = AVAILABLE_PROPERTIES[self.model]
+        _props_per_request = 15
+
+        values = self.send("get_prop", properties)
+        properties_count = len(properties)
+        values_count = len(values)
+        if properties_count != values_count:
+            _LOGGER.error(
+                "Count (%s) of requested properties does not match the "
+                "count (%s) of received values.",
+                properties_count,
+                values_count,
+            )
+        color = self.get_ambient_light()
+        return ToiletlidStatus(dict(zip(properties, values), ambient_light=color))
+
+    @command(default_output=format_output("Nozzle clean"))
+    def nozzle_clean(self):
+        """Nozzle clean."""
+        return self.send("nozzle_clean", ["on"])
```

Comment:
Should there be an "off" variant of this?

Reply:
This operation cannot be terminated